cli/config.go

```diff
@@ -65,6 +119,7 @@ type Config struct {
 	Auth            authMethod `json:"auth"`
 	LastUpdateCheck string     `json:"last_update_check"`
 	LatestVersion   string     `json:"latest_version"`
+	profile         string     `json:"-"`
```

Comment:
I haven't seen a `-` JSON struct tag before, what is this doing? I'm wondering if you don't need a struct tag at all on this field.

Reply:
The `-` struct tag says, "Don't include this field in the json representation ever, even if it's set." You can see an example of us using it [here](https://github.com/viamrobotics/rdk/blob/fb3df2ceef551aa6da9593591e32f56318096b5c/config/config.go#L696). Added because I didn't like the idea of the shape of a config changing, and in theory if someone edited the `profile` field in a config directly then it could create inconsistencies with how we parse profiled configs.

It's possible not having a struct tag at all accomplishes the same effect? I'm not sure. But the specifying seemed nice!

Reply:
Oh actually, from looking [here](https://stackoverflow.com/a/17306470) I think since `profile` is unexported this is actually totally unnecessary. Will remove!

Reply:
Cool thanks for the context! Yeah fields have to be exported to appear in JSON marshalings so keeping it private has the same effect.